requirements-cpu.txt

```diff
@@ -3,7 +3,6 @@ numpy>=1.14.0
 pygtp>=0.4
 sgf==0.5
 tqdm>=4.17
-tensorflow>=1.5,<1.6
```

Comment:
This is a bit awkward for local work. If you pip install first, shouldn't the bazel install override the tensorflow settings?

Here's another alternative: We make a requirements-base.txt. This has the shared deps from requirements-cpu and requirements-gpu.  Then we just install the base in the dockerfile and our instructions for local development are:

- install requirements-base.txt
- install either requirements-cpu.txt or requirements-gpu.txt

Reply:
negative -- if you pip install first, the final line of the install is `pip3 --no-cache-dir install --upgrade /tmp/pip/tensorflow-*.whl`, which will die as there is already a version of tensorflow installed.

Reply:
Did you see the second part of my message?

Reply:
Also, you could do pip3 uninstall I think

Reply:
I'm not sure why this is awkward for local work.  There's kind of an assumption that after `pip install -r requirements.txt` everything is ready, but that's not really necessarily true.

Anyway, if the two -cpu.txt and -gpu.txt files only contain the line 'tensorflow' vs 'tensorflow-gpu', i'd probably lean away from it in favor of having the dockerfile do 'pip uninstall'

Reply:
Oh, and actually the problem is the reverse -- with the wheel installed, `pip install -r requirements-cpu.txt` fails.

i'd rather have requirements-cpu.txt track the modules needed for the docker image, and for local  development -- everyone's kind of on their own w.r.t. TF/nvidia/openCL/anaconda/whatever

Reply:
> i'd rather have requirements-cpu.txt track the modules needed for the docker image, and for local development

If that's the case, I'd rather we have a requirements-local-cpu.txt (I can add that later). It creates a lot of friction for new users to say: Hey, here's the start of something and then debug until you figure it out. I'd rather we have a story we think works rather than one we're sure needs extra fidling

Reply:
Or, we could have a requirements-cpu-docker.txt